src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -1020,6 +1063,30 @@ export class ShowWorkspaceRecommendedExtensionsAction extends Action {
 	}
 }
 
+export class InstallWorkspaceRecommendedExtensionsAction extends Action {
+
+	static ID = 'workbench.extensions.action.installWorkspaceRecommendedExtensions';
+	static LABEL = localize('installWorkspaceRecommendedExtensions', "Install Workspace Recommended Extensions");
+
+	constructor(
+		id: string,
+		label: string,
+		@IWorkspaceContextService contextService: IWorkspaceContextService,
+		@IViewletService private viewletService: IViewletService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService
+	) {
+		super(id, label, null, contextService.hasWorkspace());
+	}
+
+	run(): TPromise<void> {
+		return this.extensionsWorkbenchService.installAllWorkspaceRecommendations();
+	}
+
+	protected isEnabled(): boolean {
+		return true;
```

Comment:
It should not be enabled if there are no recommendations or all recommended extensions are already installed and also if there is a workspace or not